### PR TITLE
[DataFrame] Add implementation for get method

### PR DESCRIPTION
**Pull request by HacksForHugs**
Add tests for get method
Add implementation/tests for get_dtype_counts method
Add implementation/tests for get_ftype_counts method

### CODE REVIEW

**Comment by robertnishihara** [comment]
ok to test

File: python/ray/dataframe/test/test_dataframe.py [hunk 1]
```diff
@@ -3,7 +3,11 @@
 from __future__ import print_function
 
 import pytest
-import ray.dataframe as rdf
+
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please remove this line.

File: python/ray/dataframe/test/test_dataframe.py [hunk 2]
```diff
@@ -3,7 +3,11 @@
...
+import sys
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please remove this line.

File: python/ray/dataframe/test/test_dataframe.py [hunk 3]
```diff
@@ -3,7 +3,11 @@
...
+sys.path.insert(0, '../../')
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please remove this line.

File: python/ray/dataframe/test/test_dataframe.py [hunk 4]
```diff
@@ -3,7 +3,11 @@
...
+import dataframe as rdf
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please revert to `import ray.dataframe as rdf`.

File: python/ray/dataframe/test/test_dataframe.py [hunk 5]
```diff
@@ -3,7 +3,11 @@
...
+
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please remove this line.

File: python/ray/dataframe/test/test_dataframe.py [hunk 6]
```diff
@@ -597,24 +601,45 @@ def test_ge():
 
 
 def test_get():
-    ray_df = create_test_dataframe()
+    df = pd.DataFrame({'col1': [0, 1, 2, 3],
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Prefer to convert test to `@pytest.fixture` and call from above. Feel free to create a new test with some string data if you would like.

File: python/ray/dataframe/dataframe.py [hunk 7]
```diff
@@ -529,13 +529,16 @@ def ge(self, other, axis='columns', level=None):
         raise NotImplementedError("Not Yet implemented.")
 
     def get(self, key, default=None):
-        raise NotImplementedError("Not Yet implemented.")
+        temp_df = self._map_partitions(lambda df: df.get(key, default=default))
```
Comment (devin-petersohn) [inline python/ray/dataframe/dataframe.py]: Please add method-level docs.

File: python/ray/dataframe/dataframe.py [hunk 8]
```diff
@@ -529,13 +529,16 @@ def ge(self, other, axis='columns', level=None):
...
+        print(type(temp_df))
+        # print(ray.get(ray_df))
```
Comment (devin-petersohn) [inline python/ray/dataframe/dataframe.py]: Please remove this code and the line above.

File: python/ray/dataframe/dataframe.py [hunk 9]
```diff
@@ -529,13 +529,16 @@ def ge(self, other, axis='columns', level=None):
...
+        return to_pandas(temp_df)
 
     def get_dtype_counts(self):
-        raise NotImplementedError("Not Yet implemented.")
+        return ray.get(_deploy_func.remote(lambda df: df.get_dtype_counts(), self._df[0]))
```
Comment (devin-petersohn) [inline python/ray/dataframe/dataframe.py]: Please add method-level docs

File: python/ray/dataframe/dataframe.py [hunk 10]
```diff
@@ -529,13 +529,16 @@ def ge(self, other, axis='columns', level=None):
...
 
     def get_ftype_counts(self):
-        raise NotImplementedError("Not Yet implemented.")
+        return ray.get(_deploy_func.remote(lambda df: df.get_ftype_counts(), self._df[0]))
```
Comment (devin-petersohn) [inline python/ray/dataframe/dataframe.py]: Please add method-level docs

File: python/ray/dataframe/test/test_dataframe.py (hunk 6 above)
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please add a test with the `default` optional parameter set to something.

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
Great first pass! All changes requested should are minor.

File: python/ray/dataframe/test/test_dataframe.py [hunk 11]
```diff
@@ -89,6 +89,8 @@ def test_copy(ray_df):
 
 @pytest.fixture
 def test_sum(ray_df, pandas_df):
+    print('ray sum', rdf.to_pandas(ray_df.sum()).sort_index())
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please remove print code.

File: python/ray/dataframe/test/test_dataframe.py [hunk 12]
```diff
@@ -89,6 +89,8 @@ def test_copy(ray_df):
...
+    print('pandas sum', pandas_df.sum())
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: Please remove print code

File: python/ray/dataframe/test/test_dataframe.py [hunk 13]
```diff
@@ -108,6 +110,21 @@ def test_transpose(ray_df, pandas_df):
     assert(ray_df_equals_pandas(ray_df.transpose(), pandas_df.transpose()))
 
 
+@pytest.fixture
+def test_get(ray_df, pandas_df, key, default=None):
```
Comment (devin-petersohn) [inline python/ray/dataframe/test/test_dataframe.py]: I prefer this test to be as follows:
```python
def test_get(ray_df, pandas_df, key):
    assert ray_df.get(key).equals(pandas_df.get(key))
    assert ray_df.get(key, default='default').equals(pandas_df.get(key, default='default'))
```
This way we can get a test with and without default but don't have to run multiple tests.

**Review by devin-petersohn — COMMENTED** [review]
This looks great! Just a couple of prints made it through, and I'd like to change something minor on the `get` test.

**Review by devin-petersohn — APPROVED** [review]
Looks great! Thanks @HacksForHugs

**Comment by devin-petersohn** [comment]
@robertnishihara @pcmoritz Is this ok to merge? There were test failures, but they seem unrelated.

**Comment by robertnishihara** [comment]
@devin-petersohn yes the test failures look unrelated. The global scheduler valgrind test seems to be failing a lot these days.

**Comment by devin-petersohn** [comment]
Thanks @HacksForHugs

**Pull request merged**